### ISSUE TITLE
Connectivity with SSHOC marketplace

### ISSUE THREAD

**Issue by proycon** [CLARIAH/clariah-plus#147]
This applies to both the FAIR Tool Discovery track and the FAIR Datasets track. Both software and data metadata may be worth propagating to the [SSHOC marketplace](https://marketplace.sshopencloud.eu/about/api-documentation).  We should investigate how.

(This addresses a point raised by @rlzijdeman during a meeting)

**Comment by rlzijdeman** [comment]
> (This addresses a point raised by @rlzijdeman during a meeting)

Something must be going well. When did it take less than 3 hours for a comment made on the board to find it's way to git :) Cheers @proycon

**Comment by proycon** [comment]
The SSHOC marketplace offers an API, documented here: https://marketplace.sshopencloud.eu/about/api-documentation      

Vocabularies and metadata guidelines are also available:
https://marketplace.sshopencloud.eu/contribute/metadata-guidelines#vocabularies
https://marketplace.sshopencloud.eu/contribute/metadata-guidelines

**Comment by proycon** [comment]
A plan has been formulated for review: https://github.com/CLARIAH/clariah-plus/blob/main/shared-development-roadmap/epics/fair-tool-discovery-sshoc-marketplace.md

**Comment by proycon** [comment]
Initial draft implementation available at https://github.com/proycon/codemeta2mp , contacted SSHOC Open Marketplace for feedback.

**Comment by rlzijdeman** [comment]
I am a bit surprised about how things 'should be done', as I thought we were exploring what would work best on both sides of the isle. Do you support the logic behind the retrieve+update approach?

**Comment by proycon** [comment]
I think it's okay to first see how far we can get with what's currently
already in place at their side. I just got access to their example repo
so I can see how they're doing it in other cases. We can then suggest
improvements if we run into obstacles. It's all still quite exploratory.

I already expressed that having a single request would be preferable
from our perspective, but that'll probably take significant effort on
their side to implement. We're keeping in contact.

**Comment by proycon** [comment]
I implemented talking to the Marketplace API, based on the feedback and examples I received before. This remains untested for now though, because I ran into issues attempting to set up a development server to test against (https://github.com/SSHOC/sshoc-marketplace-backend/issues/444).

Currently awaiting further feedback from the SSHOC side.

**Comment by proycon** [comment]
I do have enough info from SSHOC to continue this now, but what is currently unclear is the funding since CLARIAH officially ended. I know @rlzijdeman was proposing to continue this in the scope of SSHOC-NL, so I'm awaiting what comes from that before picking this up again (there is no rush anyway afaik, I'd suggest resuming this after the summer period).

**Comment by proycon** [comment]
New funding has been granted for 2025, so this will be picked up again this year (Q2?)

**Comment by proycon** [comment]
On hold pending resolution of https://github.com/SSHOC/sshoc-marketplace-backend/issues/490

**Comment by proycon** [comment]
I made some progress on this the last days,  Igot a development instance running and got a bit further with the API calls for conversion. Now it's on temporarily hold again pending https://github.com/SSHOC/sshoc-marketplace-backend/issues/494

**Comment by proycon** [comment]
Some first results of automatic conversion to the SSHOC Marketplace (dev instance):
 
* https://sshoc-marketplace-stage.acdh-dev.oeaw.ac.at/tool-or-service/4icXTQ
* https://sshoc-marketplace-stage.acdh-dev.oeaw.ac.at/tool-or-service/tGg0Hb

**Comment by proycon** [comment]
The next upcoming stage will be to seek contact with DARIAH/EOSC partners again and discuss how to proceed, a main question will be how they want to reconcile automatic and manual updates, and how to handle conflicts if tools come from multiple sources

**Comment by proycon** [comment]
I have response from the people at the marketplace-side. The bottom-line is that they're finalising implementation of a new PATCH endpoint that can deal with conflicts, some issues remain to be solved there so it's best to wait a bit. Once it is finished I will update codemeta2mp to use that new endpoint.

**Comment by proycon** [comment]
Work on this has been picked up again since early October. codemeta2mp has been modified to use PATCH instead of PUT. This remains to be tested. Things are currently pending resolution of https://github.com/SSHOC/sshoc-marketplace-backend/issues/517 .

**Comment by proycon** [comment]
Above issue was resolved, codemeta2mp released, DARIAH partners contacted for deployment in production.

**Comment by proycon** [comment]
Further discussion on production deployment proceeds in https://github.com/SSHOC/SSHOMP-Ingest/issues/54

**Comment by proycon** [comment]
Deployed in production now, ingesting data to the marketplace every sunday at 5am
